lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -407,7 +407,7 @@ sub _cancel_or_deprioritize {
             return 0;
         }
     }
-    return $job->cancel($newbuild) // 0;
+    return $job->cancel(OpenQA::Jobs::Constants::OBSOLETED) // 0;
```

Comment:
This will now always set jobs to `OBSOLETED`, before it possibly set them to `USER_CANCELLED`. So this change doesn't look correct.